### ISSUE TITLE
Add a lazy `stack` method

### ISSUE THREAD

**Issue by rafaqz** [meggart/DiskArrays.jl#158]

**Comment by meggart** [comment]
Yes, let's port this implementation to DiskArrays https://github.com/meggart/DiskArrayTools.jl/blob/d0b3f1f5cfe459cdea6c488b1272df0f185f2caf/src/DiskArrayTools.jl#L12-L75 . This will also help in deprecating DiskArrayTools, once we have moved out CFDiskArrays into its own package.
